dpki/src/seed.rs

```diff
@@ -26,10 +28,10 @@ pub enum SeedType {
     Root,
     /// Revocation seed
     Revocation,
-    /// Device specific seed
-    Device,
-    /// Derivative of a Device seed with a PIN
-    DevicePin,
+    /// Indexed seed
+    Indexed,
+    /// Derivative of a Indexed seed with a PIN
+    IndexedPin,
```

Comment:
mmm... in the whole flow, there are several steps that include indexing... so this change might be confusing.

i.e. root -> device(indexed) -> devicePin(pin hashed) -> application(indexed)

Reply:
I think there's something that I'm still fundamentally confused about regarding what was implemented in the crate and how to use it to implement the functions that are supposed to be exposed as per the spec.  I know we are using derived indexed seeds for device seeds.  But the API is allowing you to do derived indexed seeds for any purpose.  So that's why I renamed them to indexed.   And since (as far as I can tell) these functions are the only clients of the crate's functions, I thought it made sense.  But I think I may have this wrong.

Reply:
Renamed back to the old way and pulled out a common function that can be used both by the HCadmin used of the crate and Keystore.